### ISSUE TITLE
Skip hadolint DL3059 on build stages in multi-stage Dockerfile

### ISSUE THREAD

**Issue by Glenn-Terjesen** [hadolint/hadolint#1056]
- [  ] This is a bug report
- [x] This is a feature request
- [x ] I searched existing issues before opening this one

### Expected behavior
In a multi-stage Dockerfile, in the build stage(s) we want as many RUN instructions as possible to cache the steps.

https://docs.docker.com/build/building/multi-stage/

### Actual behavior
hadolint scans all parts of the file and not only the run stage.

### Steps to reproduce the behavior
Look at Dockerfile bellow.

Error: Dockerfile:11 DL3059 info: Multiple consecutive `RUN` instructions. Consider consolidation.
Error: Dockerfile:12 DL3059 info: Multiple consecutive `RUN` instructions. Consider consolidation.
Error: Dockerfile:13 DL3059 info: Multiple consecutive `RUN` instructions. Consider consolidation.

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
hadolint/hadolint-action@v3.1.0
(GitHub Actions.. )
```

**Dockerfile** (if relevant)

```Dockerfile
FROM node:18.20.4-alpine AS build
WORKDIR /usr/src/app
COPY . .
RUN npm install
RUN npm run lint
RUN CI=true npm run test
RUN npm run build

FROM node:18.20.4-alpine 
WORKDIR /usr/src/app
COPY package*.json ./
RUN npm install --only=production --ignore-scripts
COPY --from=build /usr/src/app/build .
CMD ["npm", "run", "server"]
```

**Additional environment details (OS, stack version, etc.)**

**Comment by m-ildefons** [comment]
Hi @Glenn-Terjesen,

if a linting rule doesn't apply to your particular use case, you can disable it.
This can be done globally for every Hadolint invocation through the [config file](https://github.com/hadolint/hadolint?tab=readme-ov-file#configure). It can also be customized per Docker file or even per Docker instruction with pragma comments.
In your case, adding a

```Dockerfile
# hadolint global ignore=DL3059
```

to the top of your Dockerfile would disable this rule for the entire file.
You can also override the severity of individual rules and customize the threshold for when Hadolint returns a successful or unsuccessful exit code through the config file too.


The motivation for DL3059 comes from system package managers like `apt` as they can have problems when their local package index is stale. Caching individual `RUN` instructions can cause this.

```Dockerfile
# If this is cached the next step may fail due to inconsistencies
# between the outdated info from cache and what is available on the
# repository server
RUN apt-get update

RUN apt-get install foobar
```

```Dockerfile
# This can not have inconsistencies between local package caches and
# the repository server
RUN apt-get update \
 && apt-get install foobar
```
